packages/astro/src/runtime/client/dev-toolbar/apps/audit/perf.ts

```diff
@@ -0,0 +1,126 @@
+			if (src.startsWith('data:')) return false;
+
+			// Ignore images that are smaller than 20KB, most of the time the image component won't really help with these, or they're used for specific use-cases (pixel tracking, etc.)
+			// Ignore this test for remote images for now, fetching them can be very slow and possibly dangerous
+			if (!externalUrlRe.test(src)) {
+				const imageData = await fetch(src).then((response) => response.blob());
+				if (imageData.size < 20 * 1024) return false;
+			}
+
+			return true;
+		},
+	},
+	{
+		code: 'perf-use-loading-lazy',
+		title: 'Use the loading="lazy" attribute',
+		message: (element) =>
+			`This ${element.nodeName} tag is below the fold and could be lazy-loaded to improve performance.`,
+		selector:
+			'img:not([loading]), img[loading="eager"], iframe:not([loading]), iframe[loading="eager"]',
+		match(element) {
+			const htmlElement = element as HTMLImageElement | HTMLIFrameElement;
+			// Ignore elements that are above the fold, they should be loaded eagerly
+			if (htmlElement.offsetTop < window.innerHeight) return false;
+
+			return true;
+		},
+	},
+	{
+		code: 'perf-use-loading-eager',
+		title: 'Use the loading="eager" attribute',
+		message: (element) =>
+			`This ${element.nodeName} tag is above the fold and could be eagerly-loaded to improve performance.`,
+		selector: 'img[loading="lazy"], iframe[loading="lazy"]',
+		match(element) {
+			const htmlElement = element as HTMLImageElement | HTMLIFrameElement;
+			console.log(htmlElement);
+			console.log({ offsetTop: htmlElement.offsetTop, innerHeight: window.innerHeight });
+			// Ignore elements that are below the fold, they should be loaded lazily
+			if (htmlElement.offsetTop > window.innerHeight) return false;
+
+			return true;
+		},
+	},
+	{
+		code: 'perf-use-videos',
+		title: 'Use videos instead of GIFs for large animations',
+		message:
+			'This GIF could be replaced with a video to reduce its file size and improve performance.',
+		selector: 'img[src$=".gif"]',
+		async match(element) {
+			const src = element.getAttribute('src');
+			if (!src) return false;
+
+			// Ignore remote URLs
+			if (externalUrlRe.test(src)) return false;
+
+			// Ignore GIFs that are smaller than 100KB, those are typically small enough to not be a problem
+			if (!externalUrlRe.test(src)) {
+				const imageData = await fetch(src).then((response) => response.blob());
+				if (imageData.size < 100 * 1024) return false;
+			}
+
+			return true;
+		},
+	},
+	{
+		code: 'perf-slow-component-server-render',
+		title: 'Server-rendered component took a long time to render',
+		message: (element) =>
+			`This component took an unusually long time to render on the server (${getCleanRenderingTime(
+				element.getAttribute('server-render-time')
+			)}.). This might be a sign that it's doing too much work on the server, or something is blocking rendering.`,
```

Comment:
```suggestion
			)}). This might be a sign that it's doing too much work on the server, or something is blocking the rendering.`,
```

Reply:
In my opinion, the sentence makes more sense with "blocking rendering" than "blocking the rendering", the latter naturally ask the question of "blocking the rendering of what?" whereas the former implies that it's blocking rendering in general. Could be wrong though